mcs/class/corlib/System.Globalization/CultureInfo.cs

```diff
@@ -140,7 +140,15 @@ internal static CultureInfo ConstructCurrentCulture ()
 			if (default_current_culture != null)
 				return default_current_culture;
 
-			var locale_name = get_current_locale_name ();
+			IntPtr locale_name_cstr = default (IntPtr);
+			string locale_name;
+			try {
+				locale_name_cstr = get_current_locale_name ();
+				locale_name = Marshal.PtrToStringAnsi (locale_name_cstr);
+			} finally {
+				Marshal.FreeHGlobal (locale_name_cstr); //FIXME this is wrong on windows.
+			}
```

Comment:
With what, do we have an icall that already maps to g_free? All Marshal methods map to all the weird windows alloc functions but not malloc's